src/NServiceBus.AcceptanceTests/Sagas/When_saga_is_mapped_to_complex_expression.cs

```diff
@@ -50,7 +50,10 @@ public class TestSaga02 : Saga<TestSagaData02>,
 
                 public Task Handle(OtherMessage message, IMessageHandlerContext context)
                 {
-                    Assert.AreEqual(Context.SagaId, Data.Id, "Existing instance should be found");
+                    if (Context.SagaId != Data.Id)
+                    {
+                        throw new AssertionException("Existing instance should be found");
```

Comment:
Looking at this test, concurrency is limited to 1, so would we actually be able to run into this problem? Overall it seems like it would be safe to assume each handler is invoked once.

If you think it's better to go ahead and account for concurrency anyway, I'd be OK with the collection, and then asserting that each item in the collection is the expected saga id.